job-sdk-flink/src/main/scala/org/sunbird/obsrv/job/util/BaseJobConfig.scala

```diff
@@ -0,0 +1,114 @@
+  val kafkaSystemTopic: String = config.getString("kafka.output.system.event.topic")
+  private val SYSTEM_EVENTS_OUTPUT_TAG = "system-events"
+  val systemEventsOutputTag: OutputTag[String] = OutputTag[String](SYSTEM_EVENTS_OUTPUT_TAG)
+  val systemEventsProducer = "system-events-sink"
+
+  // Checkpointing config
+  val enableCompressedCheckpointing: Boolean = if (config.hasPath("job.enable.distributed.checkpointing")) config.getBoolean("job.enable.distributed.checkpointing") else false
+  val checkpointingInterval: Int = config.getInt("task.checkpointing.interval")
+  val checkpointingPauseSeconds: Int = config.getInt("task.checkpointing.pause.between.seconds")
+  val checkpointingStateBackendType: String = if (config.hasPath("job.statebackend.type")) config.getString("job.statebackend.type") else "hashmap"
+  val enableDistributedCheckpointing: Option[Boolean] = if (config.hasPath("job.enable.distributed.checkpointing")) Option(config.getBoolean("job.enable.distributed.checkpointing")) else None
+  val checkpointingBaseUrl: Option[String] = if (config.hasPath("job.statebackend.base.url")) Option(config.getString("job.statebackend.base.url")) else None
+
+  // Base Methods
+  def inputTopic(): String
+
+  def inputConsumer(): String
+
+  def successTag(): OutputTag[T]
+
+  // Event Failures Common Variables
+  val failedEventProducer = "failed-events-sink"
+  val eventFailedMetricsCount: String = "failed-event-count"
+  val kafkaFailedTopic: String = config.getString("kafka.output.failed.topic")
+  def failedEventsOutputTag(): OutputTag[T]
+
+  def kafkaConsumerProperties(kafkaBrokerServers: Option[String] = None, kafkaConsumerGroup: Option[String] = None): Properties = {
+    val properties = new Properties()
+    properties.setProperty("bootstrap.servers", kafkaBrokerServers.getOrElse(kafkaConsumerBrokerServers))
+    properties.setProperty("group.id", kafkaConsumerGroup.getOrElse(groupId))
+    properties.setProperty(ConsumerConfig.ISOLATION_LEVEL_CONFIG, "read_committed")
+    kafkaAutoOffsetReset.map {
+      properties.setProperty("auto.offset.reset", _)
+    }
+    properties
+  }
+
+  def kafkaProducerProperties: Properties = {
+    val properties = new Properties()
+    properties.setProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaProducerBrokerServers)
+    properties.put(ProducerConfig.LINGER_MS_CONFIG, Integer.valueOf(kafkaProducerLingerMs))
+    properties.put(ProducerConfig.BATCH_SIZE_CONFIG, Integer.valueOf(kafkaProducerBatchSize))
+    properties.put(ProducerConfig.COMPRESSION_TYPE_CONFIG, kafkaProducerCompression)
+    properties.put(ProducerConfig.MAX_REQUEST_SIZE_CONFIG, Integer.valueOf(kafkaProducerMaxRequestSize))
+    properties
+  }
+
+  // String Constants
+  val CONST_OBSRV_META = "obsrv_meta"
+  val CONST_DATASET = "dataset"
+  val CONST_EVENT = "event"
+
+  // extract ets for watermark timestamping
+  private val timestampAssigner = new SerializableTimestampAssigner[mutable.Map[String, AnyRef]] {

```

Comment:
We should not hardcode the timestamp values. It should be more generic enough to handle any sort of the implementation